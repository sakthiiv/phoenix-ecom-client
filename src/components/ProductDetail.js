import React, { Component } from "react";
import {
  Card,
  CardText,
  CardBody,
  Row,
  Col
} from "reactstrap";
import cart from "../api/cart";
import sessionstorage from "../helpers/sessionstorage";
class ProductDetail extends React.Component {
  constructor() {
    super();
    this.state = {
      productAdded: false,
      successMessage:''
    }
  }
  addToCart = () => {
    const {product} = this.props.location.state
    const  user  = sessionstorage.getItemFromSession('user');

    if (user && user.userId) {
       this.addToCartCall(user.userId, product);
    }
  }

  addToCartCall = (userId, product) => {
    const productToPush = {
      "userId":userId,
      "products":[
        product
      ]
    }
   
    cart.addToCart(productToPush)
      .then(responseJson => {
        // this.props.statusRegister(true);
        console.log(responseJson.message);
        this.setState({successMessage:responseJson.message, productAdded: true});
        alert('Product added to Cart');
        window.location.reload();
      })
      .catch(error => {
        this.setState({successMessage:'some error occured'})
        console.log(error);
      });

  }
  render() {
    const { product } = this.props.location.state;
    const { productAdded,successMessage } =  this.state;
    return (
        <div className="clearfix">
            <div className="col-sm-6">  
              <picture>
                <img style={{width:'100%'}}src={"http://10.132.20.169:8080/api/v1/image/" + product.id + ".jpg"}/>
              </picture>
            </div>
            
            <div className="col-sm-6"> 
              <h1>{product.name}</h1>
              <p>{product.description}</p>
              <h2>${product.price}</h2>
              <div className="form-group">
                <input
                  type="button"
                  value="Add to cart"
                  className="btn btn-success"
                  onClick={this.addToCart}
                />
              </div>
              <p>{productAdded ? successMessage: ''}</p>
            </div>
        </div>
    );
  }
}

export default ProductDetail;
