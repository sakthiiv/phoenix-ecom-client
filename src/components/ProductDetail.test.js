import React from 'react';
import { shallow } from 'enzyme';
import ProductDetail from './ProductDetail';



const productItem = {
    "categoryId": "5bc9a87d76e3833f73511a17",
    "description": "asdasd",
    "id": "5bc9d01964151f13906681b7",
    "name": "Sony bravia",
    "price": 20000,
    "subCategoryId": "2"
};

const location = { pathname: '/product-detail', state: { product: productItem}};
describe('ProductDetail', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<ProductDetail location={location}/>);
      expect(wrapper).toMatchSnapshot();
      // On the first run of this test, Jest will generate a snapshot file automatically.
    });
});