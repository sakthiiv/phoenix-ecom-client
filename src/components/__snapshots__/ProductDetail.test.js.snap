// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetail renders correctly 1`] = `
<div
  className="clearfix"
>
  <div
    className="col-sm-6"
  >
    <picture>
      <img
        src="http://10.132.20.169:8080/api/v1/image/5bc9d01964151f13906681b7.jpg"
      />
    </picture>
  </div>
  <div
    className="col-sm-6"
  >
    <h1>
      Sony bravia
    </h1>
    <p>
      asdasd
    </p>
    <h2>
      20000
      $
    </h2>
    <div
      className="form-group"
    >
      <input
        className="btn-success"
        type="button"
        value="Add to cart"
      />
    </div>
  </div>
</div>
`;
